draft-ietf-quic-tls.md

```diff
@@ -580,6 +661,185 @@ Prior to TLS providing keys, no record protection is performed and the
 plaintext, P, is transmitted unmodified.
 
 
+## Packet Numbers {#packet-number}
+
+QUIC has a single, contiguous packet number space.  In comparison, TLS
+restarts its sequence number each time that record protection keys are
+changed.  The sequence number restart in TLS ensures that a compromise of the
+current traffic keys does not allow an attacker to truncate the data that is
+sent after a key update by sending additional packets under the old key
+(causing new packets to be discarded).
+
+QUIC does not assume a reliable transport and is required to handle attacks
+where packets are dropped in other ways.  QUIC is therefore not affected by this
+form of truncation.
+
+The packet number is not reset and it is not permitted to go higher than its
+maximum value of 2^64-1.  This establishes a hard limit on the number of packets
+that can be sent.
+
+Some AEAD functions have limits for how many packets can be encrypted under the
+same key and IV (see for example {{AEBounds}}).  This might be lower than the
+packet number limit.  An endpoint MUST initiate a key update ({{key-update}})
+prior to exceeding any limit set for the AEAD that is in use.
+
+TLS maintains a separate sequence number that is used for record protection on
+the connection that is hosted on stream 1.  This sequence number is not visible
+to QUIC.
+
+
+# Key Phases
+
+As TLS reports the availability of 0-RTT and 1-RTT keys, new keying material can
+be exported from TLS and used for QUIC packet protection.  At each transition
+during the handshake a new secret is exported from TLS and packet protection
+keys are derived from that secret.
+
+Every time that a new set of keys is used for protecting outbound packets, the
+KEY_PHASE bit in the public flags is toggled.
+
+Once the connection is fully enabled, the KEY_PHASE bit allows a recipient to
+detect a change in keying material without necessarily needing to receive the
+first packet that triggered the change.  An endpoint that notices a changed
+KEY_PHASE bit can update keys and decrypt the packet that contains the changed
+bit, see {{key-update}}.
+
+The KEY_PHASE bit on the public flags is the most significant bit (0x80).
```

Comment:
I wanted to do that in a single PR.  I was sorely tempted to do everything at once, but that PR will hit two files and I want to keep it discrete.